fails to build with LibreSSL 2.9.0

**Describe the bug**
Fails to build with LibreSSL 2.9.0. Found while testing #5049 on FreeBSD. It builds fine with LibreSSL 2.8.3 but fails with 2.9.0 due to structual change.

```
FAILED: libfreerdp/CMakeFiles/freerdp.dir/crypto/tls.c.o
/usr/bin/cc -DCMAKE_BUILD_TYPE=Release -DEXT_PATH=\"/usr/local/lib/freerdp2/extensions\" -DFREERDP_EXPORTS -DHAVE_CONFIG_H -DWINPR_EXPORTS -DWITH_OPENSSL -DWITH_WAYLAND -DWITH_X11 -DWITH_XKBFILE -D_FILE_OFFSET_BITS=64 -Dfreerdp_EXPORTS -I.
-Iinclude -I/wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/include -I/wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/winpr/include -Iwinpr/include -I/usr/local/include -O2 -pipe  -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -Wno-unused-parameter -Wno-unused-macros -Wno-padded -Wno-c11-extensions -Wno-gnu -Wno-unused-command-line-argument -Wno-deprecated-declarations -DWINPR_DLL -O2 -pipe  -fstack-protector -isystem /usr/local/include -fno-strict-aliasing -fPIC -MD -MT libfreerdp/CMakeFiles/freerdp.dir/crypto/tls.c.o -MF libfreerdp/CMakeFiles/freerdp.dir/crypto/tls.c.o.d -o libfreerdp/CMakeFiles/freerdp.dir/crypto/tls.c.o   -c /wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/libfreerdp/crypto/tls.c
/wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/libfreerdp/crypto/tls.c:1048:17: error: no member named 'alert_dispatch' in 'struct ssl3_state_st'
                tls->ssl->s3->alert_dispatch = 1;
                ~~~~~~~~~~~~  ^
/wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/libfreerdp/crypto/tls.c:1049:17: error: no member named 'send_alert' in 'struct ssl3_state_st'
                tls->ssl->s3->send_alert[0] = tls->alertLevel;
                ~~~~~~~~~~~~  ^
/wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/libfreerdp/crypto/tls.c:1050:17: error: no member named 'send_alert' in 'struct ssl3_state_st'
                tls->ssl->s3->send_alert[1] = tls->alertDescription;
                ~~~~~~~~~~~~  ^
/wrkdirs/usr/ports/net/freerdp/work/FreeRDP-2.0.0-rc4/libfreerdp/crypto/tls.c:1052:21: error: no member named 'wbuf' in
'struct ssl3_state_st'
                if (tls->ssl->s3->wbuf.left == 0)
                    ~~~~~~~~~~~~  ^
4 errors generated.
ninja: build stopped: subcommand failed.
```

--- a/libfreerdp/crypto/tls.c
+++ b/libfreerdp/crypto/tls.c
@@ -1030,7 +1030,8 @@ BOOL tls_send_alert(rdpTls* tls)
 	 * FIXME: The following code does not work on OpenSSL > 1.1.0 because the
 	 *        SSL struct is opaqe now
 	 */
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if (!defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER < 0x10100000L)) || \
+	(defined(LIBRESSL_VERSION_NUMBER) && (LIBRESSL_VERSION_NUMBER <= 0x2080300fL))
 
 	if (tls->alertDescription != TLS_ALERT_DESCRIPTION_CLOSE_NOTIFY)
 	{
@@ -1057,7 +1058,6 @@ BOOL tls_send_alert(rdpTls* tls)
 		if (tls->ssl->s3->wbuf.left == 0)
 			tls->ssl->method->ssl_dispatch_alert(tls->ssl);
 	}
-
 #endif
 	return TRUE;
 }

